Added Core files

angeryrohan opened the pull request:
This is the base layer of BassaExtension

agentmilindu commented on popup.js:
```diff
@@ -0,0 +1,138 @@
+
+document.addEventListener("DOMContentLoaded", function() {
+  var download_btn = document.getElementById("download_btn");
+  download_btn.addEventListener("click", function() {
```
You can do like, 
```suggestion
     download_btn.addEventListener("click",addDownload);
```

agentmilindu commented on popup.js:
```diff
@@ -0,0 +1,138 @@
+  iconUrl: "notificationIcon.png"
+};
+
+/*chrome.notifications.create(opt, callback);
```
Let's remove the commented out code

angeryrohan replied:
It's for future use to create a notification, Although if that's a concern, then allow me to make the code more clean by removing everything that is related to chrome notifications ^.^

agentmilindu reviewed (approved):
I'm not quite sure of the features defined in the README.md, I think we have to have a discussion on this and draw a clear roadmap. 
Code for the base of Chrome Extention LGTM

anuragparida commented on popup.js:
```diff
@@ -86,7 +86,7 @@ var n = noty({
 document.addEventListener("DOMContentLoaded", function() {
   var download_btn = document.getElementById("download_btn");
-  download_btn.addEventListener("click", function() {
+     download_btn.addEventListener("click",addDownload);
```
@agentmilindu The brackets are not closing here, I actually authored this addDownload piece of code yesterday (Rohan didn't merge it because he had already moved ahead in commits)... The event listener takes a fn as second arg, so the brackets at Line 91 should be closed.. just a gentle reminder :slightly_smiling_face:

anuragparida replied:
Oh I see @rohancl checked that in the later commit, nevermind :smile: However, when I did write that code, the server did not catch that request.. I think the "data:" token in the angular version (main Bassa) needs a corresponding token, but I could not find that in jquery so I just wrote this for partial use.. Any solutions @malithsen @vivonk @agentmilindu ?

anuragparida commented on popup.js:
```diff
@@ -89,7 +89,6 @@ document.addEventListener("DOMContentLoaded", function() {
      download_btn.addEventListener("click",addDownload);
     addDownload();
```
@rohancl Remove this function call as the event listener is calling it automatically with the argument, writing it here will make it run without the button press :slightly_smiling_face:

angeryrohan replied:
^ Yeah @anuragparida  I just noticed the new code, BassaExtension automatically shows the notification without the buttonclick, because we called that function, I've removed the function call in the updated popup.js

angeryrohan commented:
@anuragparida  @agentmilindu , I've updated the popup.js file in accordance with required changes. IN the EventListener, we only had to define a function that had to be executed on the click. Apparently, There is no need to call the function inside the body by ```addDownload();``` We could just pass it with ```download_btn.addEventListener("click",addDownload);``` . I have checked the code and it's working...

agentmilindu commented:
@malithsen @vivonk please review

vivonk commented on popup.js:
```diff
@@ -0,0 +1,115 @@
+    closeWith: ["click"],
+    open: "animated bounceInUp",
+    close: "animated bounceOutRight"
+  }
```
please lint and format your code. Maintaining good code from starting is good practice.

vivonk replied:
Let's go with 1-tab spacing instead of 2 spaces

vivonk commented on popup.js:
```diff
@@ -0,0 +1,115 @@
+    close: "animated bounceOutRight"
+  }
+});
+      $.post("http://localhost:5000/api/download/" + downurl, downurl, function(
```
make a variable where you have your base URL which is `http://localhost:5000` and then use it to any file.

vivonk replied:
``` conf.js
let basePath = 'http://localhost:5000/';
let downloadPath = basePath + 'api/download' + downlurl
$.post(downloadPath, ...)
```

angeryrohan replied:
It should be: ```let downloadPath = basePath + 'api/download/' + downlurl```

vivonk reviewed (changes requested):
Code modulation is important. Can you suggest some other structure to maintain code? Right now, it's not that good. Let's keep best practices from the first commit.

angeryrohan commented:
Sure, Absolutely agree with @vivonk . I'm on it ^.^

angeryrohan commented:
@vivonk Part-1 changes have been made, The JS code looks x10 better now :tada: 
Code Indentation of 1 tab space has been applied in popup.js.
Working on the baseURL var now...

angeryrohan commented:
Done :smile:

angeryrohan committed:
Uploaded Core files and made required changes

This is the base layer of BassaExtension
Update popup.js

Added defined function

Co-Authored-By: rohancl <rohan.clevduct@gmail.com>
Deleted readme + Removed redundant comments + Fixed a JS Bug


Update popup.js
Added CSS beautification
Code Indentation + Made modules + made better arrangements + Fixed a JS bug
Removed redundant line
Used ES6 and provided a let statement for baseURL

angeryrohan commented:
Commits Squashed. ^.^ @vivonk  Any other changes which can make Bassa Browser Extension better?
